gdpr_assist/anonymiser.py

```diff
@@ -208,21 +209,44 @@ def anonymise_field(instance, field_name):
     if cls._meta.pk.name == field_name:
         raise AnonymiseError('Cannot anonymise primary key')
 
-    # Find field
-    field = cls._meta.get_field(field_name)
-    value = getattr(instance, field_name)
+    privacy_meta = getattr(instance, GDPR_PRIVACY_INSTANCE_NAME)
+
+    if field_name in privacy_meta.fk_fields:
+        field = getattr(instance, field_name)
+
+        if field is not None:
+            instance._log_gdpr_recursive(user=user, start=True)
+            field.anonymise(user=user)
+            instance._log_gdpr_recursive(user=user, start=False)
+
+    elif field_name in privacy_meta.set_fields:
+        field = getattr(instance, field_name).all()
+        num_items = field.count()
+
+        if num_items > 0:
+            instance._log_gdpr_recursive(user=user, start=True)
+
+        [o.anonymise(user=user) for o in field]
+
```

Comment:
Since you are going to iterate over all the elements anyway, you might as well use `len` instead of `.count()` to save a query ;).

```suggestion
        related_objects = getattr(instance, field_name).all()
        num_items = len(related_objects)

        if num_items > 0:
            instance._log_gdpr_recursive(user=user, start=True)

        for related_object in related_objects:
            related_object.anonymise(user=user)

```

Reply:
This won't actually save a query. To do that, we need to turn it into a list before calling `len`. Also, I think doing things in a list comprehension is faster than a straight loop, which is why i did it that way. Want to edit your edit?

Reply:
<img width="1943" alt="IPython__picasso_server" src="https://user-images.githubusercontent.com/950449/87610153-d70d6f80-c704-11ea-8b3f-15f1e0990c36.png">

Reply:
And I think the speed difference between the for loop and the list comprehension is negligible compared to the time it'll take to perform the `anonymise` operations. IMO this is the kind of thing where readability matters more than a potential performance gain. And if you want to use speed as an argument I want to see measures to back it up ;)

Reply:
Cool! It must cache the whlle list when you call `len()` to use later. I didn't know that.

And that works for me re the loop.